pkg/cloud/gcp/actuators/machine/actuator.go

```diff
@@ -70,6 +70,8 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return a.handleMachineError(machine, fmt.Errorf(fmtErr), createEventAction)
 	}
 	if err := newReconciler(scope).create(); err != nil {
+		// Update machine and machine status in case it was modified
+		scope.Close()
```

Comment:
This is a nit; if err is a `ConflictError` then we could skip calling `scope.Close()`.